src/components/ProjectList/useProjectList.ts

```diff
@@ -1,72 +1,77 @@
 import { useNavigate } from 'react-router-dom';
+import { useEffect, useRef } from 'react';
 
+import { setList } from '@/stores/projects';
+import useInfinityScroll from '@/hooks/useInfiniteScroll';
+import { useAppSelector, useAppDispatch } from '@/stores/hooks';
+import { projectsSelector } from '@/stores/projects/selector';
+import { isLoginSelector } from '@/stores/auth';
+import { PostType } from '@/types';
+import { getChannelPosts } from '@/api/posts';
+import { inputSelector } from '@/stores/layout';
+
+export interface ProjectType {
+  _id: string;
+  image?: string;
+  name: string;
+  projectTitle: string;
+}
+const CHANNEL_ID = '6503eaffa14c752383b6a8b8';
+
+// TODO JSON.parse 에러처리, 무한스크롤 최적화, 검색,
 const useProjectList = () => {
   const navigate = useNavigate();
+  const isLogin = useAppSelector(isLoginSelector);
+  const target = useRef<HTMLDivElement>(null);
+  const dispatch = useAppDispatch();
+  const list = useAppSelector(projectsSelector);
+  const headerSearchValue = useAppSelector(inputSelector);
+
+  const { page } = useInfinityScroll({
+    target,
+    endPoint: 4,
+    options: { threshold: 0.5 },
+    list,
+  });
+
   const handleFabClick = () => {
     navigate('/projects/write');
   };
 
-  const dummyData = [
-    {
-      _id: '1',
-      name: '저쩌구씨저쩌구씨저쩌구씨저쩌구씨저쩌구씨저쩌구씨저쩌구씨저쩌구씨저쩌구씨저쩌구씨저쩌구씨저쩌구씨저쩌구씨저쩌구씨저쩌구씨저쩌구씨저쩌구씨저쩌구씨저쩌구씨저쩌구씨저쩌구씨저쩌구씨',
-      projectTitle:
-        'ProjectABCDEFGHIJKLMNOPProjectABCDEFGHIJKLMNOPProjectABCDEFGHIJKLMNOP',
-      imageUrl: 'https://source.unsplash.com/random',
-    },
-    {
-      _id: '2',
-      name: '저쩌구씨',
-      projectTitle: 'ProjectA',
-      imageUrl: 'https://source.unsplash.com/random',
-    },
-    {
-      _id: '3',
-      name: '저쩌구씨',
-      projectTitle: 'ProjectA',
-      imageUrl: 'https://source.unsplash.com/random',
-    },
-    {
-      _id: '4',
-      name: '저쩌구씨',
-      projectTitle: 'ProjectA',
-      imageUrl: 'https://source.unsplash.com/random',
-    },
-    {
-      _id: '5',
-      name: '저쩌구씨',
-      projectTitle: 'ProjectA',
-      imageUrl: 'https://source.unsplash.com/random',
-    },
-    {
-      _id: '6',
-      name: '저쩌구씨',
-      projectTitle: 'ProjectA',
-      imageUrl: 'https://source.unsplash.com/random',
-    },
-    {
-      _id: '7',
-      name: '저쩌구씨',
-      projectTitle: 'ProjectA',
-      imageUrl: 'https://source.unsplash.com/random',
-    },
-    {
-      _id: '8',
-      name: '저쩌구씨',
-      projectTitle: 'ProjectA',
-      imageUrl: 'https://source.unsplash.com/random',
-    },
-    {
-      _id: '9',
-      name: '저쩌구씨',
-      projectTitle: 'ProjectA',
-      imageUrl: 'https://source.unsplash.com/random',
-    },
-  ];
+  useEffect(() => {
+    const value = headerSearchValue.trim();
+    if (value.length < 1) return;
+    //TODO 검색 기능
+  }, [headerSearchValue]);
+
+  useEffect(() => {
+    // 두번호출떄문에 새로 업데이트
+    getChannelPosts(CHANNEL_ID, { offset: 0, limit: page * 5 + 5 })
```

Comment:
`offset`은 가져올 게시물의 초기값, `limit`은 가져올 게시물의 갯수인 것 같습니다!
제 생각에는 이렇게 해주어야 할 거 같아요!
```suggestion
    getChannelPosts(CHANNEL_ID, { offset: page * 5, limit: 5 })
```

Reply:
근데 그렇게 수정하니까 무한 스크롤이 자기 혼자서 막 불러오네요...!
일단 무한 스크롤 hook 에 뭔가 문제가 있는 것 같아요;;

Reply:
ㅠㅠ